src/entities/BaseEntity.java

```diff
@@ -0,0 +1,63 @@
+package entities;
+
+import java.time.ZonedDateTime;
+
+public abstract class BaseEntity<T> {
```

Comment:
Make T extends from Serializable

Reply:
ok i understand, serialization is used for data transmition, converting an object to bytes so that it can be easily transmited, so that it can be stored and deserialized to be used in the future